src/components/GenerativeDirectAnswerDisplay.tsx

```diff
@@ -0,0 +1,137 @@
+
+/**
+ * The CSS class interface used for {@link GenerativeDirectAnswerDisplay}.
+ *
+ * @public
+ */
+export interface GenerativeDirectAnswerDisplayCssClasses {
+  generativeDirectAnswerTitle?: string,
+  generativeDirectAnswerText: string,
+  generativeDirectAnswerDivider: string,
+  generativeDirectAnswerContainer?: string,
+  generativeDirectAnswerCitations: string,
+  generativeDirectAnswerCitation?: string,
+  generativeDirectAnswerCitationTitle?: string,
+  generativeDirectAnswerCitationSnippet?: string,
+}
+
+const builtInCssClasses: Readonly<GenerativeDirectAnswerDisplayCssClasses> = {
+  generativeDirectAnswerTitle: 'text-xl',
+  generativeDirectAnswerText: 'mt-4',
+  generativeDirectAnswerDivider: 'border-b border-gray-200 w-full pb-6 mb-6',
+  generativeDirectAnswerContainer: 'p-6 border border-gray-200 rounded-lg shadow-sm',
+  generativeDirectAnswerCitations: 'mt-4 flex overflow-x-auto gap-4',
+  generativeDirectAnswerCitation: 'p-4 border border-gray-200 rounded-lg shadow-sm bg-slate-100 flex flex-col basis-64 text-sm text-neutral',
+  generativeDirectAnswerCitationTitle: 'font-bold',
+  generativeDirectAnswerCitationSnippet: 'line-clamp-2 text-ellipsis break-words',
+};
+
+/**
+ * Props for {@link GenerativeDirectAnswerDisplay}.
+ *
+ * @public
+ */
+export interface GenerativeDirectAnswerDisplayProps<T> {
+  /** CSS classes for customizing the component styling. */
+  customCssClasses?: GenerativeDirectAnswerDisplayCssClasses
+}
+
+/**
+ * Displays the AI generated answer of a generative direct answer.
+ *
+ * @public
+ *
+ * @param props - {@link GenerativeDirectAnswerDisplayProps}
+ * @returns A React element for the generative direct answer display, or null if there is no generated answer
+ */
+export function GenerativeDirectAnswerDisplay<T>({
+  customCssClasses
+}: GenerativeDirectAnswerDisplayProps<T>): JSX.Element | null {
+  const cssClasses = useComposedCssClasses(builtInCssClasses, customCssClasses);
+
+  const isVertical = useSearchState(state => state.meta.searchType === SearchTypeEnum.Vertical);
+  const isUniversal = useSearchState(state => state.meta.searchType === SearchTypeEnum.Universal);
+  const universalResults = useSearchState(state => state.universal);
+  const verticalResults = useSearchState(state => state.vertical);
+  const searchId = useSearchState(state => state.meta.uuid);
+  const searchTerm = useSearchState(state => state.query.mostRecentSearch);
+
+  const searchResults: Result[] | undefined = React.useMemo(() => {
+    if (universalResults) {
+      return universalResults.verticals?.flatMap(v => v.results);
+    } else if (verticalResults) {
+      return verticalResults.results;
+    }
+  }, [universalResults, verticalResults]);
+
+  const searchActions = useSearchActions();
+  const gdaResponse = useSearchState(state => state?.generativeDirectAnswer?.response);
+  const isLoading = useSearchState(state => state?.generativeDirectAnswer?.isLoading);
+  
+  React.useEffect(() => {
```

Comment:
instead of using `useEffect`, can you use headless's `addListener` function to listen in changes for the results state instead?

Reply:
I have tried the `addListener` and unsure of why its callback function is triggered even when there is no state change. I'm going to try digging into search-headless to see if I can find the cause. I'll update this as soon as I figure out what's wrong with `addListener`

Reply:
this may be a bug for Search to investigate